qualtran/_infra/adjoint.py

```diff
@@ -158,9 +158,17 @@ def adjoint(self) -> 'Bloq':
         """The 'double adjoint' brings you back to the original bloq."""
         return self.subbloq
 
-    def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
+    def build_call_graph(self, ssa: 'SympySymbolAllocator') -> 'BloqCountDictT':
         """The call graph takes the adjoint of each of the bloqs in `subbloq`'s call graph."""
-        return {(bloq.adjoint(), n) for bloq, n in self.subbloq.build_call_graph(ssa=ssa)}
+        sub_cg = self.subbloq.build_call_graph(ssa=ssa)
+        if isinstance(sub_cg, dict):
+            return {bloq.adjoint(): n for bloq, n in sub_cg.items()}
+        else:
+            rtn: 'BloqCountDictT' = {}
+            for bloq, n in sub_cg:
+                adj = bloq.adjoint()
+                rtn[adj] = n + rtn.get(adj, 0)
```

Comment:
You can use `Counter`/`defaultdict(lambda: 0)` here instead of `dict`, makes the code much cleaner.

```py
counts = Counter[Bloq]()
for bloq, n in sub_cg:
    counts[bloq.adjoint()] += n
```

Reply:
Good idea, done.